v6_MIGRATION_GUIDE.md

```diff
@@ -0,0 +1,39 @@
+# Migrating to Unleash-Client-Ruby 6.0.0
+
+The upgrade the v6.0.0 of the Unleash client should be mostly seamless. There are a few significant changes that may require some changes on the consumer side or that the consumer should be generally aware of.
+
+## Custom strategy changes
+
+There's a few changes to custom strategies that may affect you if you make heavy use of them.
+
+Firstly, custom strategies are no longer allowed to override the built in strategies, namely custom strategies named 'applicationHostname', 'default', 'flexibleRollout', 'gradualRolloutRandom', 'gradualRolloutSessionId', 'gradualRolloutUserId', 'remoteAddress' or 'userWithId' will now raise an error on startup, whereas previously creating a custom strategy with one of these names would raise a warning in the logs.
+
+Secondly, the deprecated `register_custom_strategies` method has now been removed. The only way to register a custom strategy is to use configuration, i.e.
+
+```ruby
+class MyCustomStrategy
+  def name
+    'myCustomStrategy'
+  end
+
+  def is_enabled?(params = {}, context = nil)
+    true
+  end
+end
+
+Unleash.configure do |config|
+  config.strategies.add(MyCustomStrategy.new)
+end
+```
```

Comment:
This is the same as our documentation on the readme. I'm thinking if adding the snippet here adds value or if we can just link to the readme. The downside of having the snippet is that people skimming through these docs may think there's a bunch o changes, but actually, we expect users to already be doing this.

Reply:
2d9bf782e8900f72d5afbb78f4559824177632d8